scene/2d/collision_shape_2d.cpp

```diff
@@ -235,13 +244,17 @@ void CollisionShape2D::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("is_one_way_collision_enabled"), &CollisionShape2D::is_one_way_collision_enabled);
 	ClassDB::bind_method(D_METHOD("set_one_way_collision_margin", "margin"), &CollisionShape2D::set_one_way_collision_margin);
 	ClassDB::bind_method(D_METHOD("get_one_way_collision_margin"), &CollisionShape2D::get_one_way_collision_margin);
+	ClassDB::bind_method(D_METHOD("set_debug_color", "color"), &CollisionShape2D::set_debug_color);
+	ClassDB::bind_method(D_METHOD("get_debug_color"), &CollisionShape2D::get_debug_color);
 
 	ADD_PROPERTY(PropertyInfo(Variant::OBJECT, "shape", PROPERTY_HINT_RESOURCE_TYPE, "Shape2D"), "set_shape", "get_shape");
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "disabled"), "set_disabled", "is_disabled");
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "one_way_collision"), "set_one_way_collision", "is_one_way_collision_enabled");
 	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "one_way_collision_margin", PROPERTY_HINT_RANGE, "0,128,0.1"), "set_one_way_collision_margin", "get_one_way_collision_margin");
+	ADD_PROPERTY(PropertyInfo(Variant::COLOR, "debug_color"), "set_debug_color", "get_debug_color");
 }
 
 CollisionShape2D::CollisionShape2D() {
 	set_notify_local_transform(true);
+	debug_color = SceneTree::get_singleton()->get_debug_collisions_color();
```

Comment:
Should I move this to collision_shape_2d.h?

Reply:
Actually, having the default value dependent on a project setting might cause some issues if the property is saved in scenes.

Instead it could work the same way as in #46529, default to `Color(0.0, 0.0, 0.0)` and use `SceneTree::get_singleton()->get_debug_collisions_color()` for drawing when the color is equal to `Color(0.0, 0.0, 0.0)`.

I'd like to keep names consistent between nodes too, so it would be nice to call this property `debug_shape_custom_color` as well, and make a property group with `ADD_GROUP("Debug Shape", "debug_shape");` as in #46529.

Reply:
> Instead it could work the same way as in #46529, default to `Color(0.0, 0.0, 0.0)` and use `SceneTree::get_singleton()->get_debug_collisions_color()` for drawing when the color is equal to `Color(0.0, 0.0, 0.0)`.

I'm not sure if this is a clean enough solution. At a minimum, now you have to manually set the transparency for each shape, while in my solution you can simply change the hue without changing the saturation and value. And in general, magic numbers, magic strings, magic colors are bad.

I mentioned in [the comment](https://github.com/godotengine/godot-proposals/issues/2581#issuecomment-817268965) that we probably need to be able to distinguish between when the value is not set in the scene file and when the value in the file is the same as the default. In this case, if `debug_color` is not specified in the scene file, then we use the default color from the Project Settings, otherwise we use the specified value.

Reply:
IMO having the value depend on Project Settings is ok.